### PR TITLE
use yarn workspaces

### PR DESCRIPTION
**Pull request by jrhizor**
Uses Yarn and Yarn workspaces for everything.

### CODE REVIEW

File: .github/workflows/test-cli.yaml [hunk 1]
```diff
@@ -23,12 +23,13 @@ jobs:
...
           node-version: '16'
+      # todo: fix github step caching
       - name: Setup cache
         uses: actions/cache@v3
         with:
           path: |
-            ${{ github.workspace }}/cli/node_modules
-          key: ${{ runner.os }}-node-16-${{ hashFiles('cli/package-lock.json') }}
+            ${{ github.workspace }}/node_modules
+          key: ${{ runner.os }}-node-16-${{ hashFiles('yarn.lock') }}
```
Comment (tuliren) [inline .github/workflows/test-cli.yaml]: The current set up looks correct for yarn workspace.

File: .github/workflows/test-action.yaml [hunk 2]
```diff
@@ -36,7 +36,7 @@ jobs:
         with:
           path: |
             ${{ github.workspace }}/node_modules
-          key: ${{ runner.os }}-node-16-${{ hashFiles('package-lock.json') }}
+          key: ${{ runner.os }}-node-16-action-${{ hashFiles('yarn.lock') }}
```
Comment (jrhizor) [inline .github/workflows/test-action.yaml:39]: Does this scoping look right to you? Most of the data is in the root node_modules, but the action and cli builds would result in different things being installed?

File: package.json [hunk 3]
```diff
@@ -1,69 +1,7 @@
...
-  },
-  "lint-staged": {
-    "*.{ts,js,json}": [
-      "prettier --write"
-    ]
-  }
+  "private": true,
+  "workspaces": [
+    "action",
+    "cli"
```
Comment (tuliren) [inline package.json:5]: Should docs be listed as a workspace as well? Totally fine if that's meant for a separate PR.

**Review by tuliren — APPROVED** [review]
🙌

File: .github/workflows/test-action.yaml (hunk 2 above)
Comment (tuliren) [reply .github/workflows/test-action.yaml:39]: Yes, looks correct.

File: package.json (hunk 3 above)
Comment (jrhizor) [reply package.json]: I guess I can see a static dependency introduced? Added it.

**Pull request merged**